### PR TITLE
Feat/show advert status uc opp 02

### PR DESCRIPTION
**Pull request by jdijkstra-eyra**
# Pull request

## Basecamp link

...

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have not added superfluous logging
- [ ] I have added QA steps to the Basecamp ticket
- [ ] Security
  - [ ] I have considered the security implications of my changes
- [ ] Privacy
  - [ ] I have considered the privacy implications of my changes
  - [ ] I have checked new log messages for PII
  - [ ] I have added no unnecessary logging

### CODE REVIEW

File: core/systems/advert/_public.ex [hunk 1]
```diff
@@ -434,6 +432,17 @@ defmodule Systems.Advert.Public do
     end
   end
 
+  def pool_visibility(%Advert.Model{status: :online} = advert) do
+    case validate_open(advert) do
+      :ok -> :visible
+      {:error, :no_open_spots} -> :not_funded
```
Comment (mellelieuwes) [inline core/systems/advert/_public.ex]: Conflates "no open spots" with "not funded". This is actually the best end result of a study: all spots have been reserved (if spots > 0). Introduce a distinct :filled or :no_open_spots and make the user aware of this difference in the UI.

File: core/systems/advert/_public.ex [hunk 2]
```diff
@@ -434,6 +432,17 @@ defmodule Systems.Advert.Public do
...
+      {:error, :not_funded} -> :not_funded
+      {:error, :not_released} -> nil
```
Comment (mellelieuwes) [inline core/systems/advert/_public.ex:440]: Test gap for :not_released — add a test for an online advert whose submission isn't released; expect nil.

Comment (mellelieuwes) [inline core/systems/advert/_public.ex:440]: for non-:online adverts returns nil so the banner is hidden. For a researcher viewing their :concept advert, no feedback is given about why the advert isn't visible. Out of scope here, but worth a follow-up.

**Review by mellelieuwes — APPROVED** [review]
Approved with some follow-up requests
